### PR TITLE
DEV: Fix Flake8 Errors And Typos In `misc.py`

### PR DESCRIPTION
**Pull request by shaneahmed**
- Fix flake8 errors and typos in misc.py

### CODE REVIEW

File: tiatoolbox/utils/misc.py [hunk 1]
```diff
@@ -375,22 +376,60 @@ def read_locations(input_table):
...
+        """
+        if input_table.shape[1] not in [2, 3]:
+            raise ValueError("Input table must have 2 or 3 columns.")
+
+        if input_table.shape[1] == 2:
+            input_table["class"] = None
+
+        return input_table
+
     if isinstance(input_table, (str, pathlib.Path)):
```
Comment (shaneahmed) [inline tiatoolbox/utils/misc.py:425]: @John-P Any ideas on reducing cognitive complexity further?

File: tiatoolbox/utils/misc.py [hunk 2]
```diff
@@ -576,6 +607,58 @@ def unzip_data(zip_path, save_path, del_zip=True):
...
+
+def walk_list(lst):
+    """Recursive walk and jsonify a list in place.
+
+    Args:
+        lst (list):  input list.
+
+    """
+    for i, v in enumerate(lst):
+        lst[i] = walk_list_dict(v)
```
Comment (vqdang) [inline tiatoolbox/utils/misc.py:646]: I think this should be turned into a private function, we dont want to expose it to other file.

File: tiatoolbox/utils/misc.py [hunk 3]
```diff
@@ -576,6 +607,58 @@ def unzip_data(zip_path, save_path, del_zip=True):
...
+
+
+def walk_dict(dct):
```
Comment (vqdang) [inline tiatoolbox/utils/misc.py:649]: I think this should be turned into a private function, we dont want to expose it to other file.

File: tiatoolbox/utils/misc.py [hunk 4]
```diff
@@ -355,6 +358,53 @@ def contrast_enhancer(img, low_p=2, high_p=98):
     return np.uint8(img_out)
 
 
+def numpy_array_to_table(input_table):
```
Comment (vqdang) [inline tiatoolbox/utils/misc.py:361]: Should these be private function?

File: tiatoolbox/utils/misc.py [hunk 5]
```diff
@@ -355,6 +358,53 @@ def contrast_enhancer(img, low_p=2, high_p=98):
...
+        out_table["class"] = None
+        return out_table
+
+    if input_table.shape[1] == 3:
+        return pd.DataFrame(input_table, columns=["x", "y", "class"])
+
+    raise ValueError("numpy table should be of format `x, y` or " "`x, y, class`")
+
+
+def assign_unknown_class(input_table):
```
Comment (vqdang) [inline tiatoolbox/utils/misc.py:386]: Should these be private function?

File: tests/test_utils.py [hunk 6]
```diff
@@ -825,7 +825,7 @@ def test_read_point_annotations(
     assert out_table.shape[1] == 3
 
     # Test if input array does not have 2 or 3 columns
-    with pytest.raises(ValueError, match="Input array must have 2 or 3 columns"):
+    with pytest.raises(ValueError, match="numpy table should be of format"):
         _ = utils.misc.read_locations(labels_table.to_numpy()[:, 0:1])
```
Comment (vqdang) [inline tests/test_utils.py:829]: Did it trigger or error after refactoring ?

**Review by vqdang — COMMENTED** [review]
Minor points need to check, and may be changed.

File: tiatoolbox/utils/misc.py (hunk 5 above)
Comment (shaneahmed) [reply tiatoolbox/utils/misc.py]: Resolved in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/250/commits/4c2e04b86bfccdb876dd639de58b96f72885a20b

File: tiatoolbox/utils/misc.py (hunk 4 above)
Comment (shaneahmed) [reply tiatoolbox/utils/misc.py]: Resolved in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/250/commits/4c2e04b86bfccdb876dd639de58b96f72885a20b

File: tiatoolbox/utils/misc.py (hunk 3 above)
Comment (shaneahmed) [reply tiatoolbox/utils/misc.py]: Resolved in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/250/commits/4c2e04b86bfccdb876dd639de58b96f72885a20b

**Review by John-P — APPROVED** [review]

**Review by vqdang — APPROVED** [review]

**Pull request merged**